src/mscorlib/src/System/Diagnostics/Eventing/EventSource.cs

```diff
@@ -5804,6 +5822,74 @@ public enum EventManifestOptions
         AllowEventSourceOverride = 0x8,
     }
 
+#if FEATURE_EVENTSOURCE_XPLAT 
+
+    internal  class XplatEventLogger : EventListener
+    {
+        public XplatEventLogger() {}
+        
+        [DllImport(JitHelpers.QCall, CharSet = CharSet.Unicode)]
+        public static extern bool IsEventSourceLoggingEnabled();
+
+        [DllImport(JitHelpers.QCall, CharSet = CharSet.Unicode)]
+        private static extern void LogEventSource(int eventID, string eventName, string eventSourceName, string payload);
+
+        private static string Serialize(ReadOnlyCollection<string> payloadName, ReadOnlyCollection<object> payload, string sep = ", ")
+        {
+
+            var sb =  StringBuilderCache.Acquire();
+
+            if (payloadName == null || payload == null )
+                return String.Empty;
+
+            if (payloadName.Count == 0 || payload.Count == 0)
+                return String.Empty;
+
+            Contract.Assert(payloadName.Count == payload.Count);
+
+            for (int i=0; i < payloadName.Count; i++)
+            {
+                var fieldstr    = payloadName[i].ToString();
+                var valuestr = payload[i].ToString();
+
+                sb.Append(fieldstr);
+                sb.Append(":");
+                sb.Append(valuestr);
+                sb.Append(sep);
+
+            }
+
+             var result = StringBuilderCache.GetStringAndRelease(sb);
+             return  result.Substring(0,result.Length - sep.Length);
```

Comment:
Given that GetStringAndRelease is returning back a string, and strings are immutable, an you please avoid doing the substring?